compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -486,6 +532,133 @@ abstract class GenJSCode extends plugins.PluginComponent
+
+      // Generate a normal SJSDefinedJSClass
+      val origJsClass =
+        nestedGenerateClass(sym)(genScalaJSDefinedJSClass(classDef))
+
+      // Partition members into the ones we inline and the ones we keep
+      val retainedMembers = ListBuffer.empty[js.Tree]
+      var constructor: Option[js.MethodDef] = None
+      val exportedMembers = ListBuffer.empty[js.Tree]
+
+      origJsClass.defs.foreach {
+        case _: js.FieldDef => // ignore. we get it from the ctor
+
+        case mdef: js.MethodDef =>
+          mdef.name match {
+            case _: js.Ident =>
+              assert(mdef.static, "Non-static method in SJS defined JS class")
+              retainedMembers += mdef
+
+            case js.StringLiteral(name) =>
+              if (name == "constructor") {
+                assert(constructor.isEmpty, "two ctors in class")
+                constructor = Some(mdef)
+              } else {
+                exportedMembers += mdef
+              }
+          }
+
+        case property: js.PropertyDef =>
+          exportedMembers += property
+
+        case tree =>
+          sys.error("Unexpected tree: " + tree)
+      }
+
+      // Only emit static members to class
+      generatedClasses += sym -> origJsClass.copy(
+          defs = retainedMembers.toList)(
+          origJsClass.optimizerHints)(origJsClass.pos)
+
+      // Construct inline class definition
+      assert(constructor.isDefined, "No ctor found")
+      val js.MethodDef(_, _, ctorParams, _, ctorBody) = constructor.get
+
+      assert(ctorParams.size == args.size)
+      val argDefs = for ((param, arg) <- ctorParams zip args)
+        yield js.VarDef(param.name, param.ptpe, param.mutable, arg)(param.pos)
+
+      val selfName = freshLocalIdent("self")(pos)
+      def selfRef(implicit pos: ir.Position) =
+        js.VarRef(selfName)(jstpe.AnyType)
+
+      def lambda(params: List[js.ParamDef], body: js.Tree)(
+          implicit pos: ir.Position) = {
+        js.Closure(captureParams = Nil, params, body, captureValues = Nil)
+      }
+
+      val exportStats = exportedMembers.toList.map {
+        case mdef: js.MethodDef =>
+          implicit val pos = mdef.pos
+          val name = mdef.name.asInstanceOf[js.StringLiteral]
+          val impl = lambda(mdef.args, mdef.body)
+          js.Assign(js.JSBracketSelect(selfRef, name), impl)
+
+        case pdef: js.PropertyDef =>
+          implicit val pos = pdef.pos
+          val name = pdef.name.asInstanceOf[js.StringLiteral]
+          val jsObject = js.JSDotSelect(genLoadGlobal(), js.Ident("Object"))
+
+          def accessor(tree: js.Tree, name: String)(genBody: js.Tree => js.Tree) =
+            if (tree == js.EmptyTree) Nil
+            else List(js.StringLiteral(name) -> genBody(tree))
+
+          val accessorObj = js.JSObjectConstr(
+              accessor(pdef.getterBody, "get")(lambda(params = Nil, _)) ++
+              accessor(pdef.setterBody, "set")(lambda(pdef.setterArg :: Nil, _))
+          )
```

Comment:
You should also set `configurable` and `enumerable` to `true`, to be consistent with the simple `=` for methods.